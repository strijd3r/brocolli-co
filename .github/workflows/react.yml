# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: Node.js CI

on:
  push:
  pull_request:

jobs:
  build:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14.x]

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Set up Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}

      - name: Install dependencies
        run: yarn

      - name: Run the tests
        run: yarn test

      - name: Build
        run: yarn build
        if: startsWith(github.ref, 'refs/tags/')

      - name: Release
        uses: softprops/action-gh-release@v1
        if: startsWith(github.ref, 'refs/tags/')
        with:
          files: |
            README.md
            package.json
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      - name: Deploy
        uses: peaceiris/actions-gh-pages@v3
        if: startsWith(github.ref, 'refs/tags/')
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./build
