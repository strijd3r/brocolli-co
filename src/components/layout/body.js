import React, { Component } from 'react'
import styled from 'styled-components'

/**
 * BodyComponent
 * @param {object} props BodyComponent properties
 * @param {string} props.children Children
 * @param {string} props.className Classname
 * @returns
 */
function BodyComponent({ children, className }) {
    return (
        <section className={className}>{children}</section>
    )
}

/**
 * Body is a styled version of the BodyComponent.
 * @type {Component}
 */
export const Body = styled(BodyComponent)`
flex: 1;
`