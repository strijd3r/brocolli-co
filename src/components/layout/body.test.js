import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import { Body } from './body';

test('renders div and its children', () => {
  const body = 'testing react body';

  const { container } = render(<Body>{body}</Body>);
  expect(container.firstChild).toMatchSnapshot();

  const bodyEl = screen.getByText(body);
  expect(bodyEl).toBeInTheDocument();
});
