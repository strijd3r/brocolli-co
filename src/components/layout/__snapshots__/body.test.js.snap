// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders div and its children 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<section
  class="c0"
>
  testing react body
</section>
`;
